src/main/java/reactor/rabbitmq/OutboundMessageResult.java

```diff
@@ -19,9 +19,9 @@
 /**
  * Result of a sent message when using publisher confirm.
  */
-public class OutboundMessageResult {
+public class OutboundMessageResult<T extends OutboundMessage> {
 
-    private final OutboundMessage outboundMessage;
+    private final T outboundMessage;
```

Comment:
Users that don't use special `OutboundMessage` extensions should still be able to use untyped/raw `OutboundMessageResult` in backward compatible fashion